gitbutler-app/src/main.rs

```diff
@@ -98,48 +98,74 @@ fn main() {
+                    app_handle.manage(users_storage_controller.clone());
 
-                    let zipper = zip::Controller::try_from(&app_handle)
-                        .expect("failed to initialize zipc controller ");
-                    tauri_app.manage(zipper);
+                    let users_controller = users::Controller::new(users_storage_controller.clone());
+                    app_handle.manage(users_controller.clone());
 
-                    let deltas_controller = deltas::Controller::try_from(&app_handle).expect("failed to initialize deltas controller");
-                    app_handle.manage(deltas_controller);
+                    let projects_controller = projects::Controller::new(
+                        app_data_dir.clone(),
+                        projects_storage_controller.clone(),
+                        users_controller.clone(),
+                        Some(watcher_controller.clone())
+                    );
+                    app_handle.manage(projects_controller.clone());
+
+                    app_handle.manage(assets::Proxy::new(app_cache_dir.join("images")));
+
+                    let database_controller = database::Database::open(app_data_dir.join("database.sqlite3")).expect("failed to open database");
+                    app_handle.manage(database_controller.clone());
 
-                    let sessions_controller = sessions::Controller::try_from(&app_handle)
-                        .expect("failed to initialize sessions controller");
-                    app_handle.manage(sessions_controller);
+                    let zipper = zip::Zipper::new(&app_cache_dir);
+                    app_handle.manage(zipper.clone());
 
-                    let projects_controller = projects::Controller::try_from(&app_handle)
-                        .expect("failed to initialize projects controller");
-                    app_handle.manage(projects_controller);
+                    app_handle.manage(zip::Controller::new(app_data_dir.clone(), app_log_dir.clone(), zipper.clone(), projects_controller.clone()));
 
-                    let vbranch_contoller =
-                        virtual_branches::controller::Controller::try_from(&app_handle)
-                            .expect("failed to initialize virtual branches controller");
-                    app_handle.manage(vbranch_contoller);
+                    let deltas_database_controller = deltas::database::Database::new(database_controller.clone());
+                    app_handle.manage(deltas_database_controller.clone());
 
-                    let keys_controller = keys::Controller::try_from(&app_handle).expect("failed to initialize keys controller");
-                    app_handle.manage(keys_controller);
+                    let deltas_controller = deltas::Controller::new(deltas_database_controller.clone());
+                    app_handle.manage(deltas_controller);
```

Comment:
Variable is only used here, could drop the declaration like you've done in some places?

Reply:
Yeah I'll clean it up tomorrow. :)